File: implementations/sdk/java/src/test/java/com/spearsystems/lcp/SchemaValidatorTest.java
package com.spearsystems.lcp;

import org.junit.jupiter.api.Test;
import java.nio.file.Path;

final class SchemaValidatorTest {
    @Test
    void validatesCanonicalLeadFixture() throws Exception {
        var root = Path.of("../../../");
        var validator = SchemaValidator.fromDirectory(root.resolve("schemas"));
        validator.validateEnvelope(java.nio.file.Files.readString(root.resolve("examples/lead.json")));
    }
}

File: implementations/sdk/kotlin/src/test/kotlin/com/spearsystems/lcp/SchemaValidatorTest.kt
package com.spearsystems.lcp

import java.nio.file.Path
import kotlin.test.Test

class SchemaValidatorTest {
    @Test
    fun validatesCanonicalLeadFixture() {
        val root = Path.of("../../../")
        val validator = LcpSchemaValidator.fromDirectory(root.resolve("schemas"))
        validator.validateEnvelope(root.resolve("examples/lead.json").toFile().readText())
    }
}
